app/assets/javascripts/admin/customers/directives/edit_address_dialog.js.coffee

```diff
@@ -25,13 +26,18 @@ angular.module("admin.customers").directive 'editAddressDialog', ($compile, $tem
       else
         scope.addressType = 'ship_address'
       scope.address = scope.customer[scope.addressType]
-      scope.states = scope.filter_states(scope.address?.country_id)
+      scope.states = scope.filterStates(scope.address?.country_id)
 
       template = $compile($templateCache.get('admin/edit_address_dialog.html'))(scope)
       template.dialog(DialogDefaults)
       template.dialog('open')
       scope.$apply()
 
-    scope.filter_states = (countryID) ->
+    scope.filterStates = (countryID) ->
       return [] unless countryID
-      $filter('filter')(scope.availableCountries, {id: parseInt(countryID)}, true)[0].states
\ No newline at end of file
+      $filter('filter')(scope.availableCountries, {id: parseInt(countryID)}, true)[0].states
+
+    scope.clearInvalidState = ->
```

Comment:
I think a name like `clearStateOfAnotherCountry` or something similar would better convey what this function is meant to do. Did I understand it correctly?

Reply:
I am a splitter, so I would do:
```coffee
scope.clearState() unless scope.addressStateMatchesCountry()

...

scope.clearState = ->
  scope.address.state_id = ""

scope.addressStateMatchesCountry = ->
  scope.states.some (state) -> state.id == scope.address.state_id
```

Reply:
That's even better! +1

Reply:
all right, looks great. thanks a lot guys! done.